### PR TITLE
add rpc to query raft status

### PR DESCRIPTION
**Pull request by kikimo**
#### What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### What does this PR do?

This PR add an readonly raft rpc `getState()` to query raft info like: leadership, running status, current term and etc. We expect this rpc to help us find the correct leader as soon as possible when leader change happends during our test, it's a replacement for meta api which has high latency and sometime return incorrect leader information.

### CODE REVIEW

**Review by liuyu85cn — DISMISSED** [review]
LGTM

File: src/interface/raftex.thrift [hunk 1]
```diff
@@ -7,6 +7,21 @@ namespace cpp nebula.raftex
...
+                     // except it does not participate in leader election
+}
+
+enum Status {
+    STARTING            = 0; // The part is starting, not ready for service
+    RUNNING             = 1; // The part is running
+    STOPPED             = 2; // The part has been stopped
+    WAITING_SNAPSHOT    = 3; // Waiting for the snapshot.
+}
+
```
Comment (yixinglu) [inline src/interface/raftex.thrift:24]: These enum types have been defined in following source file, could you unify the usage of these enum types?

https://github.com/vesoft-inc/nebula/blob/b7c690114016ce855df000abcc376f8903b30b12/src/kvstore/raftex/RaftPart.h#L239-L252

Comment (kikimo) [reply src/interface/raftex.thrift:24]: @yixinglu updated, PTAL

File: src/interface/raftex.thrift [hunk 2]
```diff
@@ -7,6 +7,21 @@ namespace cpp nebula.raftex
 
 include "common.thrift"
 
+enum Role {
+    LEADER      = 1, // the leader
+    FOLLOWER    = 2; // following a leader
+    CANDIDATE   = 3; // Has sent AskForVote request
+    LEARNER     = 4; // same with FOLLOWER, except that it does
+                     // not vote in leader election
+}
```
Comment (critical27) [inline src/interface/raftex.thrift:16]: could add a `(cpp.enum_strict)` here

**Review by critical27 — COMMENTED** [review]
Others LGTM

**Review by critical27 — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]
LGTM

**Pull request merged**